src/nasa_provider.js

```diff
@@ -0,0 +1,56 @@
+import axios from "axios";
+
+const createRequest = (path, params) => {
+    let request = `${process.env.MARS_PHOTOS_URL}${path}`;
+
+    if (!params) {
+        params = {};
+    }
+    params.api_key = process.env.API_KEY;
+
+    request += '?';
+    for (const [key, value] of Object.entries(params)) {
+        request += `${key}=${value}&`
+    }
+
+    return request;
+};
+
+const rovers = async () => {
+    let response = undefined;
+    try {
+        response = await axios.get(createRequest('/rovers'));
+    } catch (e) {
+        console.error(e);
+    }
+
+    return response.data;
```

Comment:
I think this will error if we get an error from axios. I think we can move this retun line inside the try (and remove the let response = undefined;). We should also then throw our own error inside the catch. Something like 'Get /rovers request returned non 200 status'